### PR TITLE
Add watched lots to Inbox 'bids' tab

### PR DESCRIPTION
**Pull request by erikdstock**
The type of this PR is: **feature**

### Description
This PR resolves [PURCHASE-2206]. It is based off of and closes #4344. It adds watched lots to a user's active bids inbox 🎉
cc @artsy/purchase-devs 

Note: A regression prior to this PR broke the pull-refresh mechanism again. We should prioritize a fix for it but it is not new to this change

### PR Checklist (tick all before merging)

- [x] **Analytics included**
- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

In this GIF: 
- i am not registered for the first sale, only watching a lot
- i am registered for the second sale, bid on lot 1, watching lot 2, bidding and watching lot 6.

[PURCHASE-2206]: https://artsyproduct.atlassian.net/browse/PURCHASE-2206

[CX-434]: https://artsyproduct.atlassian.net/browse/CX-434

### CODE REVIEW

**Review by jo-rs — APPROVED** [review]
It's a big boy!
With the exception of the refresh, which we've already discussed, it seems to be working great! 💯 
I'm still combing through the code, but on a first, second, and third glance everything seems to be looking good. I think it could be good to revisit the whole `My Bids` section in the future, and brainstorm some refactoring, but nothing blocking atm.

**Review by erikdstock — COMMENTED** [review]
Some comments prior to cleanup.

File: src/lib/Scenes/MyBids/Components/BiddingStatuses.tsx [hunk 1]
```diff
@@ -49,3 +49,13 @@ export const Passed = () => (
     Passed
   </Text>
 )
+
+export const Watching = () => (
+  <>
+    <BookmarkFill />
+    <Text variant="caption" color="black60">
+      {" "}
```
Comment (xtina-starr) [inline src/lib/Scenes/MyBids/Components/BiddingStatuses.tsx:57]: Minor but don't we need the correct icon here. Might be worth adding a ToDo or creating a ticket to pull the icon into palette so we don't forget.

**Comment by xtina-starr** [comment]
This looks great @erikdstock. Thanks for taking this over.

File: src/lib/Scenes/MyBids/Components/BiddingStatuses.tsx (hunk 1 above)
Comment (erikdstock) [reply src/lib/Scenes/MyBids/Components/BiddingStatuses.tsx:57]: `BookmarkFill` is the correct icon 🙆‍♂️

Comment (erikdstock) [reply src/lib/Scenes/MyBids/Components/BiddingStatuses.tsx:57]: image

**Pull request merged**